core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/datastore/DataFile.java

```diff
@@ -121,29 +121,61 @@ public long storeData(byte[] data) throws IOException {
 		return offset;
 	}
 
+	// This variable is used for predicting the number of bytes to read in getData(long offset). This helps us to only
+	// need to execute a single IO read instead of first one read to find the length and then one read to read the data.
+	int dataLengthApproximateAverage = 25;
```

Comment:
Very clever. Is 25 as a starting point based on any sort of observation of what is "typical"? Or is it more a "we need to start somewhere"?